### PR TITLE
Simplify ByteStringStreamSource, cleanup

### PR DESCRIPTION
**Pull request by werkt**
ByteStringStreamSource is not used in any capacity which requires the
ByteSource interface/impl. Clean up some syntax and indicate thread
interruption on InterruptedException catch during stream operations.

### CODE REVIEW

File: src/main/java/build/buildfarm/instance/memory/ByteStringStreamSource.java [hunk 1]
```diff
@@ -45,15 +44,15 @@ public void write(byte[] b) {
 
       @Override
```
Comment (ola-rozenfeld) [inline src/main/java/build/buildfarm/instance/memory/ByteStringStreamSource.java]: now that it no longer extends anything, you need to remove the @Override from everywhere

Comment (werkt) [reply src/main/java/build/buildfarm/instance/memory/ByteStringStreamSource.java]: and I am unimpressed that that did not warn me! thanks!

Comment (werkt) [reply src/main/java/build/buildfarm/instance/memory/ByteStringStreamSource.java]: Ah, now I recall, note that this is in the anonymous override of InputStream and OutputStream (@Override does error in the line 75 case below

Comment (ola-rozenfeld) [reply src/main/java/build/buildfarm/instance/memory/ByteStringStreamSource.java]: Ah, right, sorry!

**Pull request merged**